lib/synapse/service_watcher/marathon.rb

```diff
@@ -0,0 +1,105 @@
+require "synapse/service_watcher/base"
+require 'json'
+require 'resolv'
+
+module Synapse
+  class MarathonWatcher < BaseWatcher
+    def start
+      @check_interval = @discovery['check_interval'] || 10.0
+      @connection = nil
+      @watcher = Thread.new { watch }
+    end
+
+    def stop
+      @connection.finish
+    rescue
+      # pass
+    end
+
+  private
+
+    def validate_discovery_opts
+      required_opts = %w[marathon_api_url application_name]
+
+      required_opts.each do |opt|
+        if @discovery.fetch(opt, '').empty?
+          raise ArgumentError,
+            "a value for services.#{@name}.discovery.#{opt} must be specified"
+        end
+      end
+    end
+
+    def attempt_marathon_connection
+      @marathon_api = URI.join(@discovery['marathon_api_url'], "/v2/apps/#{@discovery['application_name']}/tasks")
```

Comment:
Idk? It seems pretty stable to me, and I think even if they were to make a breaking change to it they might still maintain the old URL. Nonetheless, I will make it configurable so I don't have to eat my words.